datanode/entities/stop_orders.go

```diff
@@ -172,6 +172,13 @@ func StopOrderFromProto(so *pbevents.StopOrderEvent, vegaTime time.Time, seqNum
 		ocoLinkID = StopOrderID(*so.StopOrder.OcoLinkId)
 	}
 
+	if so.StopOrder.ExpiresAt != nil {
+		expiresAt = ptr.From(NanosToPostgresTimestamp(*so.StopOrder.ExpiresAt))
+		if expiresAt.Before(vegaTime) {
```

Comment:
Can we remove that if? That would return an error once the stop order get updated at the time it expires